modules/nf-core/modules/samtools/index/main.nf

```diff
@@ -16,10 +16,17 @@ process SAMTOOLS_INDEX {
     tuple val(meta), path("*.crai"), optional:true, emit: crai
     path  "versions.yml"           , emit: versions
 
+    when:
+    task.ext.when == null || task.ext.when
+
     script:
     def args = task.ext.args ?: ''
     """
-    samtools index -@ ${task.cpus-1} $args $input
+    samtools \\
+        index \\
+        -@ ${task.cpus-1} \\
```

Comment:
Separate issue but why are the modules written like this? The scheduler already allocates `task.cpus` and then the process just wastes the available cores.

Reply:
It was added in https://github.com/nf-core/modules/pull/1069 @FriederikeHanssen @phue @maxulysse .